src/app/clusters/basic-information/basic-information.cpp

```diff
@@ -279,22 +298,20 @@ CHIP_ERROR BasicAttrAccess::Read(const ConcreteReadAttributePath & aPath, Attrib
     }
 
     case SpecificationVersion::Id: {
-        status = ReadSpecificationVersion(aEncoder);
+        InteractionModelEngine::GetInstance()->GetDataModelProvider()->GetNodeDataModelConfiguration(nodeConfig);
+        status = aEncoder.Encode(nodeConfig.specVersion);
         break;
     }
 
     case MaxPathsPerInvoke::Id: {
-        status = ReadMaxPathsPerInvoke(aEncoder);
+        InteractionModelEngine::GetInstance()->GetDataModelProvider()->GetNodeDataModelConfiguration(nodeConfig);
```

Comment:
Always do this at the very top, as it's cheap and probably needed. Alternatively, make a separate switch statement at the top that calls `GetNodeDataModelConfiguration` for the right cases without needing to put this in every sub-case.

Reply:
Understood, I figured since only these 4 attributes reads needed it, it was worth moving into the cases, but I will just move it up 👍🏻